internal/magic/text.go

```diff
@@ -348,3 +352,35 @@ func Srt(in []byte, _ uint32) bool {
 	// A third line must exist and not be empty. This is the actual subtitle text.
 	return s.Scan() && len(s.Bytes()) != 0
 }
+
+// Vtt matches a Web Video Text Tracks (WebVTT) file. See
+// <https://www.iana.org/assignments/media-types/text/vtt>.
+func Vtt(raw []byte, limit uint32) bool {
+	sigStarts := [][]byte{
+		{0xEF, 0xBB, 0xBF, 0x57, 0x45, 0x42, 0x56, 0x54, 0x54}, // UTF-8 BOM and "WEBVTT"
+		{0x57, 0x45, 0x42, 0x56, 0x54, 0x54},                   // "WEBVTT"
+	}
+	sigEnds := []*byte{
+		bytePointer(0x0A), // line feed
+		bytePointer(0x0D), // carriage return
+		bytePointer(0x20), // space
+		bytePointer(0x09), // horizontal tab
+		nil,               // EOF
+	}
+
+	for _, sigStart := range sigStarts {
+		for _, sigEnd := range sigEnds {
+			if sigEnd == nil {
+				if bytes.Equal(raw, sigStart) {
+					return true
+				}
+				continue
+			}
+			sig := append(sigStart, *sigEnd)
+			if bytes.HasPrefix(raw, sig) {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
I think simple is better.  
```go
func Vtt(raw []byte, _ uint32) bool {
	prefixes := [][]byte{
		{the},
		{eight},
		{sequences},
		{without},
		{EOF},
	}

	for _, p := range prefixes {
		if bytes.HasPrefix(raw, p) {
			return true
		}
	}

	return bytes.Equal(raw, []byte{first EOF sequence}) ||
		bytes.Equal(raw, []byte{second EOF sequence})
}
```

Reply:
Resolved with af85bd637819ac4da4543e81cb5b920db876a489.